lib/fika/type_checker.ex

```diff
@@ -231,24 +223,24 @@ defmodule Fika.TypeChecker do
   # Atom value
   def infer_exp(env, {:atom, _line, atom}) do
     Logger.debug("Atom value found. Type: #{atom}")
-    {:ok, ":#{atom}", env}
+    {:ok, %T.Atom{value: atom}, env}
```

Comment:
I've also introduced the Atom type so this ":#{..}" interpolation is automatically used instead of the default interpolation for atoms